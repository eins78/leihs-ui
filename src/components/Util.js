import React, { Fragment as F } from 'react'
import f from 'lodash'

export const Let = ({ children, ...p }) => <F>{children(p)}</F>

export const breakLinesReact = text =>
  // 'hello\nworld' => hello<br/>world
  !!text &&
  (f.isArray(text) ? text : text.split('\n')).map((item, i, a) => (
    <React.Fragment key={i}>
      {item}
      {/* no break after last line: */}
      {i + 1 < a.length && <br />}
    </React.Fragment>
  ))
