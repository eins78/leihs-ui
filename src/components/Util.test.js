import React, { Fragment as F } from 'react'
import renderer from 'react-test-renderer'

import { Let, breakLinesReact } from './Util'

const examples = [
  {
    name: 'Let',
    content: (
      <F>
        {/* TODO: <Let /> should fail*/}
        <Let a="1" b={2}>
          {args => JSON.stringify(args)}
        </Let>
      </F>
    )
  },
  {
    name: 'breakLinesReact',
    content: (
      <F>
        <p>{breakLinesReact()}</p>
        <p>{breakLinesReact('')}</p>
        <p>{breakLinesReact('hello')}</p>
        <p>{breakLinesReact('hello\\nworld')}</p>
        <p>{breakLinesReact('hello\nworld \\n \netc\np\tp')}</p>
      </F>
    )
  }
]

examples.forEach(({ name, content }) => {
  it(`${name}: renders correctly`, () => {
    expect(renderer.create(content).toJSON()).toMatchSnapshot()
  })
})
