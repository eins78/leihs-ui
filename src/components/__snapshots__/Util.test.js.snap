// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Let: renders correctly 1`] = `"{\\"a\\":\\"1\\",\\"b\\":2}"`;

exports[`breakLinesReact: renders correctly 1`] = `
Array [
  <p />,
  <p />,
  <p>
    hello
  </p>,
  <p>
    hello\\nworld
  </p>,
  <p>
    hello
    <br />
    world \\n 
    <br />
    etc
    <br />
    p	p
  </p>,
]
`;
